tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir

```diff
@@ -370,6 +370,16 @@ func.func @test_logical_not(%arg0: tensor<1x21x3xi1>) -> tensor<*xi1> {
   func.return %0 : tensor<*xi1>
 }
 
+// -----
+// CHECK-LABEL: test_reduce_all
+// CHECK-DAG: %[[VAR0:.*]] = "tosa.reduce_all"(%arg0) {axis = 0 : i64}
+// CHECK: %[[VAR1:.*]] = "tosa.reshape"(%[[VAR0]]) {new_shape = [21, 3]}
```

Comment:
You can elide this as this is part of the common lowering and should hopefully be tested directly rather than specific to reduce_all lowering. (else I'd want to see this for a couple of different axis and with result type of the reduce_all to make it clear why reshape is needed).

Reply:
Fair enough. I've replaced that one by some lowering tests with different axis and the keep_dim option.